packages/css/src/util/fetch/PausableFetcher.ts

```diff
@@ -0,0 +1,55 @@
+import { getLoggerFor } from '@solid/community-server';
+import type { StatusDependant } from './StatusDependant';
+import type { FetchParams, Fetcher } from './Fetcher';
+import EventEmitter from 'events';
+
+const PROCESS = Symbol();
+
+/**
+ * A {@link Fetcher} wrapper that is {@link StatusDependant}, 
+ * enabling requests to be backlogged during inactive periods.
+ */
+export class PausableFetcher implements Fetcher, StatusDependant<boolean> {
+  protected readonly logger = getLoggerFor(this);
+  protected readonly backlog: EventEmitter[] = [];
+  protected active = false;
+  protected processing = false;
+
+  constructor(
+    protected readonly fetcher: Fetcher,
+  ) {}
+
+  async fetch(...args: FetchParams): Promise<Response> {
+    const trigger = new EventEmitter();
+    const result = new Promise<Response>(resolve => {
+      trigger.on(PROCESS, () => {
+        this.fetcher.fetch(...args).then(response => resolve(response))
```

Comment:
```suggestion
        this.fetcher.fetch(...args).then(resolve)
```

Reply:
🤔 is this necessary, or are you instinctively applying ESlint rules? 😄

Reply:
It just felt wrong to not at least mention it 😉